7.x 2.x panels

sherakama opened the pull request:
## Not Ready for pull yet.

Opening up a pull request so you can see the work as it is progressing and our initial updates. More to come.m

sherakama committed:
Adding Git Ignore File and Panel Templates

sherakama committed:
--append

sherakama committed:
Added Plugins to info file

Commit:
Rearranging and updating the theme structure to pull out bootstrap and font-awesome files into their own directories (css, img, js)

Commit:
Updating path in font awesome

Commit:
File structure for parent theme

Commit:
Adding the various less files for the sub theme css

Commit:
Adding default jquery code to script.js

Commit:
Minor documentation change

Commit:
updating documentation in readme file

Commit:
Adding panels layouts to sub-theme

Commit:
Updating the one-column panels layout

Commit:
Panels layout themeing - 3 column layout

Commit:
Using mixins for the inline-blocks

Commit:
Final styling changes for 3x3x3 panels layout

Commit:
Merge branch 'panels_layouts' into 7.x-2.x-panels

Commit:
Merge branch 'file_structure' into 7.x-2.x-panels

Commit:
Panels 2 column 50-50 layout

Commit:
Merge branch 'panels-2col_50_50' into 7.x-2.x-panels

Commit:
Panels layouts for Panels 33/66 and 66/33 layouts

Commit:
Panels layouts for Panels 25-25-50, 25-50-25, 50-25-25, layouts

Commit:
Adding files for the 66/33 & 4 x 25

Commit:
Creating the new panels layout

Commit:
Final changes to the 66/33 x 4-25% Grid

Commit:
Adding the 33/66 - 4x25% layout

Commit:
Adding the 2 x 3 column layout

Commit:
Minor fixes to get the 2 x 3 column grid working

Commit:
Adding the 33-66 X 50-25-25 panels layout to the grid-ready category

Commit:
Merge branch 'panels_2x3' into 7.x-2.x-panels

Commit:
Merge branch 'panels_2x4' into 7.x-2.x-panels

Commit:
Merge branch 'panels_2-col_66_33' into 7.x-2.x-panels

Commit:
Merge branch 'panels-3col_25_50_25' into 7.x-2.x-panels

sherakama committed:
Merge pull request #7 from thejmsill/panels_2x3

Panels 2x3

sherakama committed:
Merge pull request #2 from thejmsill/file_structure

File structure

sherakama committed:
Merge pull request #3 from thejmsill/panels-2col_50_50

Panels 2col 50 50

sherakama committed:
Merge pull request #4 from thejmsill/panels_2-col_66_33

Panels 2 col 66 33

sherakama committed:
Merge pull request #5 from thejmsill/panels-3col_25_50_25

Panels 3col 25 50 25

sherakama committed:
Merge pull request #6 from thejmsill/panels_2x4

Panels 2x4

sherakama committed:
Merge branch '7.x-2.x' of github.com:SU-SWS/open_framework into swsupdates

Commit:
Updating .info file with correct CSS path
Adding the Panels layout to the base theme.
Removing the panels layout folder from the sub-theme.

sherakama committed:
Fixed some formatting issues

sherakama committed:
Merge pull request #8 from thejmsill/panels-config

Adjustments to Theme Structure

Commit:
Merge branch '7.x-2.x-panels' of github.com:imagex/open_framework into 7.x-2.x-panels

Commit:
Adding a 1-2-2-1 column layout (used in the header mini-panel)

hyperboy committed:
Updated to jquery-migrate-1.2.1.min.js
#143

Commit:
Configuring the layout struture for the panel - 1, 33/66, 66/33, 1

Commit:
Adding the 1-6-1-2-1 panels layout to the OFW grid ready panels category

Commit:
Styling and template changes to ensure the regions have a full width option available

sherakama committed:
Merge pull request #9 from thejmsill/panels-1-2-2-1

Panels 1 2 2 1

sherakama committed:
Merge pull request #10 from thejmsill/panels-1-6-1-2-1

Panels 1 6 1 2 1

Commit:
Updating the panel images to demonstrate the layouts

Commit:
Updating the panel images to demonstrate the layouts - With Better images

Commit:
Adding a five column layout to the panels system

sherakama committed:
Merge pull request #11 from thejmsill/five-column

Five column

meganem commented:
Awesome! We'd love at some point to see a demo of the changes and the panels templates if you're up for it!

Commit:
Fixed panel custom template (sites/all/themes/open_framework/plugins/layouts/grid-ready/one_two_two_one) removing extra % from css

hyperboy committed:
Removed duplicate region references in subtheme_kit.info.txt
#146

hyperboy committed:
Resolved undefined index: scope in drupal_get_js() issue.
#147

Commit:
Adding the a 1-2-1-2-4 layout (100% - 50/50 - 100 - 50/50 - 16/16/16/50)

Commit:
Minor tweaks to the panels layout 1-2-1-2-4

sherakama committed:
Merge pull request #12 from thejmsill/variable-1-2-2-4

Adding the a 1-2-1-2-4 layout (100% - 50/50 - 100 - 50/50 - 16/16/16/50)

Commit:
Adding a new panels layout (1-2-4-6-1-2-1)

Commit:
Adding conditional items to the first and second rows

sherakama committed:
Merge pull request #13 from thejmsill/panels_var_complex1

Panels var complex1

sherakama committed:
Merge branch '7.x-2.x' into 7.x-2.x-panels

Commit:
Updating the two_33_66_4x25 layout as the panels configuration wasn't css'd correctly

sherakama committed:
Merge pull request #14 from thejmsill/panel-update

Updating the two_33_66_4x25 layout as the panels configuration wasn't cs...

Commit:
Adding some float clears to fix the layout shifting in panels

sherakama committed:
Merge pull request #15 from thejmsill/float-fix

Adding some float clears to fix the layout shifting in panels

Commit:
SOEDEPT-169 Update jquery ui to be compatible with jquery.

Open Framework manually overrides the jquery version to use
jquery 1.9.1. Doing so makes the jquery ui version that
ships with Drupal incompatible. As a result, Open Framework
also needs to override the jquery ui in order to prevent
various jquery ui related functionalities from breaking.

Includes jquery ui 1.10.3 and overrides jquery ui js and css.
Uses jquery update module as a reference point for how to
do this, but does not follow the same implementation because
it appears that hook_library_alter() cannot be used inside
a theme.

Down the line if this approach proves bug-ridden, it would be
better to either downgrade the jquery version back to 1.8 or
lower to not break compatibility with the jquery ui version
in core OR force Open Framework to depend on jquery_update
module.

sherakama committed:
Merge pull request #16 from kevinchampion/jquery_ui_conflict

SOEDEPT-169 Update jquery ui to be compatible with jquery.

sherakama committed:
Fixed PHP Notice Errors Galore!

meganem commented:
Hey guys, what's your timeframe for this? We are sprinting this week on OFW udpates, so just checking through the issue queue to prioritize for the week. Let us know when you think this might be ready for testing on our end?

sherakama commented:
This is working pretty well on our end now. We can probably rebase this PR into one nice commit for you and merge it into your branch. You can also directly pull our current branch on to your local machine if you would like to test it out. @thejmsill is off this week but we should probably schedule a meeting with him as well to go over the layouts and additions to this version. 

I will get Michael to set up a meeting. A few of our team members will be on site at campus this week.

meganem commented:
Awesome! Would love a demo from you guys whenever is convenient.

We'll need to do some testing on our end against our current client sites in production so just let us know when you have it ready and we can get that set up.

sherakama commented:
I would suggest keeping it as a separate branch. You can keep this 'flavour' up to date along with the 7.x-2.x branch with simple git merges as it was originally branched from the 7.x-2.x branch. The same ancestor will allow for changes to both quite easily.

Codelica commented:
:+1:  Please do consider keeping Panels integration as a separate branch, or included via some configuration option.  While Panels are popular, there are still Drupal purists out there who like to keep things as native and lightweight as possible. :)

sherakama commented:
Panels is not 'heavy' as far as performance goes. Actual testing has shown that it panels is quicker in most cases: http://www.codeenigma.com/en/blog/performance-panels

The templates provided in this theme won't really get in the way but its additional clutter for someone who doesn't want them.

Cheers,
Shea

Codelica commented:
Versus Blocks (especially Views constructed Blocks) I won't argue. :)  But when sticking to core's template system to eeak out performance, memory use, etc. I think it's still worth consideration.  Thanks.

tassaf85 commented:
Hello everybody, thank you very much for your great work.. When you will finish Panels support for OFW, I need to use it soon, can you tell me a date please?

tassaf85 commented:
Any update please?

sherakama commented:
@tassaf85,

The pull request is pretty stable at this point. If you want to use this code you can clone the SU-SWS panels branch on to your project and then add ImageX as a second remote repository. Once you have added ImageX as a second remote pull down the panels branch from the ImageX remote into your repository. Later on when this pull request gets merged and any future updates happen you will be fine to pull down from just the SU-SWS repository. 

I hope this makes sense. 

Cheers,

tassaf85 commented:
Thanks for your reply, but I need your help
I am not familiar with github, so can you send me the file to my email tassaf85@gmail.com

Thank you

sherakama commented:
I would highly advise that you take some time to learn more about git and github. In the mean time this should work for you.

https://github.com/imagex/open_framework/archive/7.x-2.x-panels.zip

DannyJoris committed:
removed content and content-wrapper ids from one.tpl.php to prevent duplication

woombo committed:
Merge pull request #17 from DannyJoris/duplicate-content-ids

removed content and content-wrapper ids from one.tpl.php to prevent duplication

tassoman commented:
It could works on 2.0.6 release? I've seen your activity was don in May, 2013 but latest tag was released by SU-SWS in october...

sherakama commented:
This PR has gone stale. Going to pull it into the panel branch and start new issue threads from there.
